Assets/Tests/InputSystem/CoreTests_Controls.cs

```diff
@@ -1119,6 +1119,63 @@ public void Controls_CanFindControlsUsingWildcards_InMiddleOfNames()
         }
     }
 
+    [Test]
+    [Category("Actions")]
+    public void Actions_BindingMask_HandlesWildcards()
+    {
+        var keyboard = InputSystem.AddDevice<Keyboard>();
+
+        var action = new InputAction();
+        action.AddBinding("<Keyboard>/a");
+        action.Enable();
+
+        using (var trace = new InputActionTrace())
+        {
+            trace.SubscribeTo(action);
+
+            Assert.That(action.controls, Is.EquivalentTo(new[] { keyboard.aKey }));
+
+            // Enable only keyboards via mask
+            action.bindingMask = new InputBinding(path: "<Keyboard>/*");
+
+            Assert.That(action.controls, Is.EquivalentTo(new[] { keyboard.aKey }));
+
+            Press(keyboard.aKey);
+            Release(keyboard.aKey);
+
+            Assert.That(trace,
+                Started(action, keyboard.aKey)
+                    .AndThen(Performed(action, keyboard.aKey))
+                    .AndThen(Canceled(action, keyboard.aKey)));
+        }
+    }
+
+    [Test]
+    [Category("Controls")]
+    public void Controls_MatchingBindingToMask()
+    {
+        var mask = new InputBinding("<Keyboard>/e");
+        var binding = new InputBinding("<Keyboard>/e");
+        Assert.That(mask.MatchesMask(ref binding), Is.True);
+        mask.path = "<Keyboard>/d";
+        Assert.That(mask.MatchesMask(ref binding), Is.False);
+        mask.path = "<Keyboard>/*";
+        Assert.That(mask.MatchesMask(ref binding), Is.True);
+        mask.path = null;
```

Comment:
I'd be tempted to split some out into a new tests to improve readability

This is a good guide on test naming
https://internaldocs.unity.com/automation/Guidelines/naming/
UnitOfWork_WhenStateUnderTest_ProducesExpectedBehavior()

Perhaps some like:
Controls_BindingWithoutMask_Matches
Controls_BindingMaskWithSpecificKey_IgnoresOtherKey
Controls_BindingMaskWithWildcard_Matches
Controls_BindingMaskWithNullPath_Ignored
Controls_BindingMaskWithNullName_Ignored
Controls_BindingMaskWithMousePath_FiltersOutKeyboard